src/machine/machine_stm32f407.go

```diff
@@ -101,15 +81,16 @@ func enableAltFuncClock(bus unsafe.Pointer) {
 
 // UART
 type UART struct {
-	Buffer *RingBuffer
-	Bus    *stm32.USART_Type
+	Buffer          *RingBuffer
+	Bus             *stm32.USART_Type
+	AltFuncSelector stm32.AltFunc
 }
 
 var (
-	// Both UART0 and UART1 refer to USART2.
 	UART0 = UART{
-		Buffer: NewRingBuffer(),
-		Bus:    stm32.USART2,
+		Buffer:          NewRingBuffer(),
+		Bus:             stm32.USART2,
+		AltFuncSelector: stm32.AF7_USART1_2_3,
```

Comment:
These are somewhat dependent on the pins used, so ideally they'd be in the `UARTConfig` struct, but for now they're probably fine here; things just won't work if there's a mismatch between pins, bus and alt-func selected.

Reply:
Hmm that extra field is unfortunate but okay for now.

Reply:
yeah i couldn't come up with a better way to pass the alternate function parameter around; it's used on a number of the STM32 families, but even from one family to another the way it's set up varies :(